File: src/main/java/um/haberes/core/hexagonal/liquidaciones/acreditacion_pago/domain/model/AcreditacionPago.java
package um.haberes.core.hexagonal.liquidaciones.acreditacion_pago.domain.model;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AcreditacionPago {

    private Long acreditacionPagoId;

    private Integer anho;

    private Integer mes;

    private OffsetDateTime fechaPago;

    private BigDecimal totalSantander;

    private BigDecimal totalOtrosBancos;

    private Integer comprobanteIdPago;

    private Integer puntoVentaPago;

    private Long numeroComprobantePago;
}

File: src/main/java/um/haberes/rest/kotlin/model/AcreditacionPago.kt
package um.haberes.rest.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import um.haberes.rest.model.Auditable
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["anho", "mes", "fechaPago"])])
data class AcreditacionPago(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var acreditacionPagoId: Long? = null,

    var anho: Int = 0,
    var mes: Int = 0,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaPago: OffsetDateTime? = null,
    var totalSantander: BigDecimal = BigDecimal.ZERO,
    var totalOtrosBancos: BigDecimal = BigDecimal.ZERO,
    var comprobanteIdPago: Int? = null,
    var puntoVentaPago: Int? = null,
    var numeroComprobantePago: Long? = null

) : Auditable()
